src/Octoshift/Services/VersionChecker.cs

```diff
@@ -21,29 +22,27 @@ public VersionChecker(HttpClient httpClient, OctoLogger log)
 
     public async Task<bool> IsLatest()
     {
-        var curVersion = GetCurrentVersion();
+        var currentVersion = GetCurrentVersion();
         var latestVersion = await GetLatestVersion();
 
-        curVersion = curVersion[..latestVersion.Length];
-
-        return latestVersion.CompareTo(curVersion) <= 0;
+        return currentVersion.CompareTo(latestVersion) != -1;
     }
 
-    public string GetCurrentVersion()
+    public Version GetCurrentVersion()
     {
-        return Assembly.GetExecutingAssembly().GetName().Version.ToString();
+        return Assembly.GetExecutingAssembly().GetName().Version;
```

Comment:
In order to prevent any confusion, additionally we can return the first 3 segments of the version:
```c#
Assembly.GetExecutingAssembly().GetName().Version.ToString(3);
```